### PR TITLE
Support for text-only prom endpoints who speaks http correctly + newer dd-agent

### PR DESCRIPTION
**Pull request by mumoshu**
# TL;DR;

Configure the prometheus check with `format: text` in case you encounter `HTTPError: 406 Client Error: Not Acceptable for url`.

# PROBLEM

The prometheus check had been trying to http-get a prom endpoint assuming it is serving PROTOBUF format,
even though it was actually serving TEXT format, which in some cases resulted in errors like:

> HTTPError: 406 Client Error: Not Acceptable for url: http://10.2.93.28:8080/metrics

More concretely, omitting the kwarg `pFormat` of `self.poll` had been resulting to the format `protobuf` rather than `text`, which results in adding an `Accept: application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited` header to your http request to a prom endpoint. A correctly implemented http server can reject the request due to mismatch between the `Accept` request header vs actual content-type of the prom metrics to be responded.

See https://prometheus.io/docs/instrumenting/exposition_formats/ for definitions of PROTOBUF and TEXT formats and corresponding Content-Type.

A real example of such correctly implemented text-only prom endpoints is one implemented with https://github.com/prometheus/client_ruby#rack-middleware.
It rejects requests w/ the above Accept header with 406 errors because the implementation specifically supports the TEXT format only.

# FIX

This change allows you to configure which format to be used specifically by PrometheusCheck, so that you can avoid the error.

You could write your config like:

```
init_config:

instances:
  - target: http://localhost:9100/metrics
    format: text
    config:
      drop:
        - .*
      keep:
        - node_cpu.*
        - node_memory.*
```

to avoid 406s in this case.

### CODE REVIEW

**Comment by mumoshu** [comment]
Note: If you're encountering errors like:

```
2018-01-15 07:33:15 UTC | ERROR | dd.collector | checks.prometheus(__init__.py:812) | Check 'prometheus' instance #0 failed
Traceback (most recent call last):
  File "/opt/datadog-agent/agent/checks/__init__.py", line 795, in run
    self.check(copy.deepcopy(instance))
  File "/etc/dd-agent/checks.d/prometheus.py", line 69, in check
    instance=instance)
  File "/etc/dd-agent/checks.d/prometheus.py", line 49, in process
    content_type, data = self.poll(endpoint, headers=headers)
  File "/opt/datadog-agent/agent/checks/prometheus_check.py", line 302, in poll
    req.raise_for_status()
  File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/requests/models.py", line 862, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 406 Client Error: Not Acceptable for url: http://10.2.93.28:8080/metrics
```

This is the fix you need.

**Comment by mumoshu** [comment]
Also note that this change doesn't require you to modify your config as long as you're not seeing the 406 errors. Do specify `format: text` when and only when you are seeing them.

**Comment by mumoshu** [comment]
@discordianfish Hi - thanks for maintaining this project! Would you mind reviewing my change?
Probably this isn't affecting you yet but would be a good-to-have to prevent errors in the future.

**Comment by discordianfish** [comment]
Hi @mumoshu,

it would be great if we could get content-negotiation working instead of having to set that flag. What do you think?

For the dd-agent stuff, that can be merged as-is if you submit a separate PR for that.

**Comment by discordianfish** [comment]
Well, let's merge this for now and add content negotiation later.

**Comment by mumoshu** [comment]
@discordianfish Thanks for the review! Yea, good point - I'll try to add that later.

At glance PrometheusCheck seemed like forcing me to specify a PrometheusFormat but after reading [code](https://github.com/DataDog/dd-agent/blob/f519ebe18998c173442988c6ce6f28d7833e7a6e/checks/prometheus_check.py#L399-L402), I came up with a easy work-around.

I believe we can call the poll method like:

```python
headers['accept-encoding'] = 'gzip'
headers['accept'] = 'text/plain; version=0.0.4, ' \
                                'application/vnd.google.protobuf; ' \
                                'proto=io.prometheus.client.MetricFamily; ' \
                                'encoding=delimited'
self.poll(endpoint, headers=headers, pFormat=None)
```

so that it's all up to a http server implementation to actually do content negotiation.
This is such a small change because PrometheusCheck [parses the response according to the content-type](https://github.com/DataDog/dd-agent/blob/f519ebe18998c173442988c6ce6f28d7833e7a6e/checks/prometheus_check.py#L141), regardless of what we set for pFormat, surprisingly :)
